internal/api/keppel/accounts_test.go

```diff
@@ -300,40 +272,51 @@ func TestAccountsAPI(t *testing.T) {
 				"name":           "second",
 				"auth_tenant_id": "tenant1",
 				"gc_policies":    gcPoliciesJSON,
-				"in_maintenance": false,
-				"metadata":       assert.JSONObject{},
 				"rbac_policies":  rbacPoliciesJSON,
 			},
 		},
 	}.Check(t, h)
 	tr.DBChanges().AssertEqual(`
-		INSERT INTO accounts (name, auth_tenant_id, metadata_json) VALUES ('first', 'tenant1', '{"bar":"barbar","foo":"foofoo"}');
+		INSERT INTO accounts (name, auth_tenant_id) VALUES ('first', 'tenant1');
 		INSERT INTO accounts (name, auth_tenant_id, gc_policies_json, rbac_policies_json) VALUES ('second', 'tenant1', '[{"match_repository":".*/database","except_repository":"archive/.*","time_constraint":{"on":"pushed_at","newer_than":{"value":10,"unit":"d"}},"action":"protect"},{"match_repository":".*","only_untagged":true,"action":"delete"}]', '[{"match_repository":"library/.*","permissions":["anonymous_pull"]},{"match_repository":"library/alpine","match_username":".*@tenant2","permissions":["pull","push"]}]');
 	`)
 
 	// check editing of InMaintenance flag (this also tests editing of GC policies
 	// since we don't give any and thus clear the field)
-	for _, inMaintenance := range []bool{true, false} {
+	for _, isDeleting := range []bool{true, false} {
+		account := assert.JSONObject{
+			"auth_tenant_id": "tenant1",
+			"rbac_policies":  rbacPoliciesJSON,
+		}
+		accountExcept := assert.JSONObject{
+			"name":           "second",
+			"auth_tenant_id": "tenant1",
+			"rbac_policies":  rbacPoliciesJSON,
+		}
+		if isDeleting {
+			account = assert.JSONObject{
+				"auth_tenant_id": "tenant1",
+				"state":          "deleting",
```

Comment:
How is this test supposed to work? In any case, writing to `state` in the PUT endpoint should be rejected.